File: arrayHash/leetcode_443/solution.py
# from collections import defaultdict
# class Solution:
#     def compress(self, chars: list[str]) -> int:
#         word_dic = defaultdict(int)
#         s = ""
#         for c in chars:
#             word_dic[c] += 1

#         chars = []
#         for key in word_dic.keys():
#             if word_dic[key] == 1:
#                 s = s + key
#                 chars.append(key)
#             else:
#                 s = s + key + str(word_dic[key])
#                 chars.append(key)
#                 chars.append(str(word_dic[key]))
#         return chars

File: arrayHash/leetcode_443/solution.rb
# @param {Character[]} chars
# @return {Integer}
def compress(chars)
    i = 0
    write = 0

    while i < chars.length
        char = chars[i]
        count = 0
        while i < chars.length and char == chars[i]
            i += 1
            count += 1
        end

        chars[write] = char
        write += 1

        if count > 1
            for count_str in count.to_s.split("")
                chars[write] = count_str
                write += 1
            end
        end
    end

    return write

end